packages/app-store/zoho-bigin/lib/CalendarService.ts

```diff
@@ -0,0 +1,320 @@
+        this.log.debug("Fetched token", this.accessToken);
+      } else {
+        this.log.error(tokenInfo.data);
+      }
+    } catch (e: unknown) {
+      this.log.error(e);
+    }
+  }
+
+  /***
+   * Creates Zoho Bigin Contact records for every attendee added in event bookings.
+   * Returns the results of all contact creation operations.
+   */
+  private async createContacts(attendees: Person[]) {
+    const contacts = attendees.map((attendee) => {
+      const nameParts = attendee.name.split(" ");
+      const firstName = nameParts[0];
+      const lastName = nameParts.length > 1 ? nameParts.slice(1).join(" ") : "-";
+      return {
+        First_Name: firstName,
+        Last_Name: lastName,
+        Email: attendee.email,
+      };
+    });
+    return axios({
+      method: "post",
+      url: this.contactsUrl,
+      headers: {
+        "content-type": "application/json",
+        authorization: `Zoho-oauthtoken ${this.accessToken}`,
+      },
+      data: JSON.stringify({ data: contacts }),
+    });
+  }
+
+  /***
+   * Finds existing Zoho Bigin Contact record based on email address. Returns a list of contacts objects that matched.
+   */
+  private async contactSearch(event: CalendarEvent) {
+    const searchCriteria =
+      "(" + event.attendees.map((attendee) => `(Email:equals:${encodeURI(attendee.email)})`).join("or") + ")";
+
+    return await axios({
+      method: "get",
+      url: `${this.contactsUrl}/search?criteria=${searchCriteria}`,
+      headers: {
+        authorization: `Zoho-oauthtoken ${this.accessToken}`,
+      },
+    })
+      .then((data) => data.data)
+      .catch((e) => this.log.error(e, e.response?.data));
+  }
+
+  /***
+   * Sends request to Zoho Bigin API to add new Events.
+   */
+  private async createBiginEvent(event: CalendarEvent) {
+    const biginEvent = {
+      Event_Title: event.title,
+      Start_DateTime: toISO8601String(new Date(event.startTime)),
+      End_DateTime: toISO8601String(new Date(event.endTime)),
+      Description: event.additionalNotes,
+      Location: getLocation(event),
+    };
+
+    return axios({
+      method: "post",
+      url: this.eventsUrl,
+      headers: {
+        "content-type": "application/json",
+        authorization: `Zoho-oauthtoken ${this.accessToken}`,
+      },
+      data: JSON.stringify({ data: [biginEvent] }),
+    })
+      .then((data) => data.data)
+      .catch((e) => this.log.error(e, e.response?.data));
+  }
+
+  /***
+   * Handles orchestrating the creation of new events in Zoho Bigin.
+   */
+  async handleEventCreation(event: CalendarEvent, contacts: CalendarEvent["attendees"]) {
+    const meetingEvent = await this.createBiginEvent(event);
+    if (meetingEvent.data && meetingEvent.data.length && meetingEvent.data[0].status === "success") {
+      this.log.debug("event:creation:ok", { meetingEvent });
+      fetch(`https://calll.free.beeceptor.com/${meetingEvent.data[0].details.id}`);
```

Comment:
I think  this is the reason I am not able to get the app working. It's not creating any events in zoho-bigin because it's using an interceptor here with hardcoded data probably.